testing/tests/DevExpress.viz.core/baseAxis.tests.js

```diff
@@ -1199,6 +1200,43 @@ QUnit.test('Validate visualRange, option is set', function(assert) {
     assert.deepEqual(this.axis.getOptions().visualRange, [new Date(10), new Date(20)]);
 });
 
+QUnit.module('Deprecated options', {
+    beforeEach() {
+        environment.beforeEach.apply(this, arguments);
+        sinon.spy(errors, 'log');
+    },
+    afterEach() {
+        environment.afterEach.apply(this, arguments);
+        errors.log.restore();
+    },
+    createAxis() {
+        return new Axis({
+            renderer: this.renderer,
+            axisType: 'xyAxes',
+            drawingType: 'linear',
+            isArgumentAxis: false,
+            eventTrigger: () => { },
+            getTemplate() {}
+        });
+    }
+});
+
+QUnit.test('Should show warning if deprecated "shift" value sets to "valueAxis.visualRangeUpdateMode" option', function(assert) {
+    const axis = this.createAxis();
+    axis.updateOptions({ label: {}, visualRangeUpdateMode: 'shift' });
+
+    axis.validate();
+
+    assert.deepEqual(errors.log.lastCall.args,
+        [
+            'W0016',
+            'valueAxis.visualRangeUpdateMode',
+            'shift',
+            '23.1',
+            'Specify another value'
+        ]);
+});
+
```

Comment:
Deprecation message is logged twice on option runtime change. Let's fix this and add a test